packages/juce-blueprint/src/components/Canvas.ts

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+
+export function bindCanvasContextProperties(ctx: any) {
+  Object.defineProperty(ctx, 'fillStyle', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setFillStyle(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'strokeStyle', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setStrokeStyle(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'lineWidth', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setLineWidth(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'font', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setFont(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'textAlign', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setTextAlign(value);
+    }
+  });
+}
+
+export interface CanvasProps {
+  onMeasure?: (width: number, height: number) => void;
+  onDraw: (ctx: CanvasRenderingContext2D) => void;
+  autoclear?: boolean;
+}
+
+interface CanvasState {
+  width: number;
+  height: number;
+}
+
+export class Canvas extends Component<CanvasProps, CanvasState> {
+  constructor(props: CanvasProps) {
+    super(props);
+
+    this._onMeasure = this._onMeasure.bind(this);
+    this._onDraw = this._onDraw.bind(this);
+
+    this.state = {
+      width: 0,
+      height: 0
+    };
+  }
+
+  _onMeasure(e: any) {
+    this.setState({
+      width: e.width,
+      height: e.height
+    });
+
+    if (typeof this.props.onMeasure === 'function') {
+      this.props.onMeasure(e.width, e.height);
+    }
+  }
+
+  _onDraw(ctx: any) {
```

Comment:
It's possible to type this against CanvasRenderingContext2D from lib dom. Worth doing now do you think? Or should we wait till we can provide the full API (or at least the majority of it)?

Reply:
Awesome, yea I think big +1 for typing it as `CanvasRenderingContext2D`, but that's a cool idea about the `BlueprintCanvasRenderingContext2D`. Maybe we start with the latter, and as we build out support for the remaining methods we make a plan to switch to the standard `CanvasRenderingContext2D` type def?

Reply:
Yeah that sounds sensible to me. I leave it for the moment and add that as a follow up if that's cool. I think we probably want an `add-additional-typings` PR which brings in all the ViewProps etc.